### PR TITLE
ANDROID-14388 Add support for Tu brand

### PR DESCRIPTION
**Pull request by yamal-alm**
### :goal_net: What's the goal?
Create new "Tu" brand.

:warning: Don't panic 🙂, most of files are autogenerated styles and screenshots. I tried to add a comment on every file :warning:

### :construction: How do we do it?
- Required boilerplate was added to existing code so tokens generator could work with corresponding [`tu.json` file from mistica-design](https://github.com/Telefonica/mistica-design/blob/production/tokens/tu.json) repo.
- I then triggered GitHub job and generate Tu tokens
- Autogenerated #333 branch was merged into current one (I didn't wait for any reviewer approval, everything is being reviewed altogether in this PR for simplicity)
- Added required boilerplate to screenshot test suites
- Screenshots baseline was updated with Tu brand after having added it to test suites
- I also wrote some documentation about how to add a new brand

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
- [x] 🖼️ Screenshots/Videos
- [x] [Mistica App download link](https://install.appcenter.ms/orgs/tuenti-organization/apps/mistica-catalog/distribution_groups/public/releases/435)
- [x] Reviewed by Mistica design team

https://github.com/Telefonica/mistica-android/assets/47142570/ccc1b08c-419e-4890-992d-73570b9a1b6d

### CODE REVIEW

File: library/src/test/java/com/telefonica/mistica/button/ButtonTest.kt [hunk 1]
```diff
@@ -95,13 +96,13 @@ internal class ButtonTest(
     }
 }
 
-fun Brand.getBaseThemeForBrand() = when (this) {
+fun Brand.getBaseThemeForBrand(): Int = when (this) {
     MovistarBrand -> R.style.MisticaTheme_Movistar_test
     VivoBrand -> R.style.MisticaTheme_Vivo_test
     VivoBrand -> R.style.MisticaTheme_O2_test
     VivoBrand -> R.style.MisticaTheme_Blau_test
```
Comment (yamal-alm) [inline library/src/test/java/com/telefonica/mistica/button/ButtonTest.kt:103]: this is a bug in tests already reported https://jira.tid.es/browse/ANDROID-14391

File: library/src/test/res/values/styles.xml [hunk 2]
```diff
@@ -104,5 +104,19 @@
 		<item name="linkButtonInverse_test">@style/AppTheme.Button.LinkButtonInverse</item>
 	</style>
 
+	<style name="MisticaTheme.Tu.test" parent="MisticaTheme.Tu">
```
Comment (yamal-alm) [inline library/src/test/res/values/styles.xml:107]: more boilerplate required for xml screenshot testing

File: library/src/main/res/values/themes_vivo_autogenerated.xml [hunk 3]
```diff
@@ -62,7 +62,7 @@
 		<item name="colorTextLinkSnackbar">@color/vivo_color_vivoPurpleLight50</item>
 		<item name="colorControl">@color/vivo_color_grey3</item>
 		<item name="colorControlActive">@color/vivo_color_vivoPurple</item>
-		<item name="colorControlInverse">@color/vivo_color_white</item>
+		<item name="colorControlInverse">@color/vivo_color_vivoPurpleLight50</item>
```
Comment (yamal-alm) [inline library/src/main/res/values/themes_vivo_autogenerated.xml:65]: unrelated token update

File: library/src/main/res/values/themes_tu.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+	<style name="MisticaTheme.Tu_Customizations" parent="MisticaTheme.Tu_Base">
```
Comment (yamal-alm) [inline library/src/main/res/values/themes_tu.xml:3]: required manual boilerplate for Tu xml theme. These are default values already used in other brands

File: library/src/main/res/values/themes_o2_autogenerated.xml [hunk 5]
```diff
@@ -62,7 +62,7 @@
 		<item name="colorTextLinkSnackbar">@color/o2_color_o2BluePrimary30</item>
 		<item name="colorControl">@color/o2_color_grey3</item>
 		<item name="colorControlActive">@color/o2_color_o2BluePrimary</item>
-		<item name="colorControlInverse">@color/o2_color_white</item>
+		<item name="colorControlInverse">@color/o2_color_o2BluePrimary30</item>
```
Comment (yamal-alm) [inline library/src/main/res/values/themes_o2_autogenerated.xml:65]: unrelated token update

File: library/src/main/res/values/themes_blau_autogenerated.xml [hunk 6]
```diff
@@ -62,7 +62,7 @@
 		<item name="colorTextLinkSnackbar">@color/blau_color_blauPurple30</item>
 		<item name="colorControl">@color/blau_color_grey2</item>
 		<item name="colorControlActive">@color/blau_color_blauBlueSecondary</item>
-		<item name="colorControlInverse">@color/blau_color_white</item>
+		<item name="colorControlInverse">@color/blau_color_blauBluePrimary30</item>
```
Comment (yamal-alm) [inline library/src/main/res/values/themes_blau_autogenerated.xml:65]: unrelated token update

File: library/src/main/res/values/colors_tu_autogenerated.xml [hunk 7]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (yamal-alm) [inline library/src/main/res/values/colors_tu_autogenerated.xml:2]: new autogenerated Tu color palette

File: library/src/main/res/values/attrs_components.xml [hunk 8]
```diff
@@ -313,6 +313,7 @@
             <enum name="telefonica" value="3" />
             <enum name="blau" value="4" />
             <enum name="vivoNew" value="5" />
+            <enum name="tu" value="6" />
```
Comment (yamal-alm) [inline library/src/main/res/values/attrs_components.xml:316]: more manuel boilerplate

File: library/src/main/res/values-night/themes_tu_autogenerated.xml [hunk 9]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (yamal-alm) [inline library/src/main/res/values-night/themes_tu_autogenerated.xml:2]: dark mode autogenerated style

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/VivoNewBrandColorsAutogenerated.kt [hunk 10]
```diff
@@ -65,7 +65,7 @@ public object VivoNewBrandColors {
       textLinkSnackbar = VivoNewPaletteColor.vivoNew_color_vivoPurpleLight20,
       control = VivoNewPaletteColor.vivoNew_color_grey4,
       controlActivated = VivoNewPaletteColor.vivoNew_color_vivoPurple,
-      controlInverse = VivoNewPaletteColor.vivoNew_color_white,
+      controlInverse = VivoNewPaletteColor.vivoNew_color_vivoPurpleLight50,
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/VivoNewBrandColorsAutogenerated.kt:68]: unrelated token update

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/TuBrandFontSizesAutogenerated.kt [hunk 11]
```diff
@@ -0,0 +1,11 @@
+// Code generated automatically, DO NOT EDIT manually. Use Import design tokens GitHub Action.
+package com.telefonica.mistica.compose.theme.brand
+
+import androidx.compose.ui.unit.TextUnit
+import androidx.compose.ui.unit.sp
+
+public object TuBrandFontSizes {
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/TuBrandFontSizesAutogenerated.kt:7]: autogenerated

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/TuBrandColorsAutogenerated.kt [hunk 12]
```diff
@@ -0,0 +1,349 @@
+// Code generated automatically, DO NOT EDIT manually. Use Import design tokens GitHub Action.
+package com.telefonica.mistica.compose.theme.brand
+
+import androidx.compose.ui.graphics.Color
+import com.telefonica.mistica.compose.theme.color.MisticaColors
+
+public object TuBrandColors {
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/TuBrandColorsAutogenerated.kt:7]: autogenerated compose color palette

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/TelefonicaBrandColorsAutogenerated.kt [hunk 13]
```diff
@@ -71,7 +71,7 @@ public object TelefonicaBrandColors {
       textLinkSnackbar = TelefonicaPaletteColor.telefonica_color_telefonicaBlue30,
       control = TelefonicaPaletteColor.telefonica_color_grey3,
       controlActivated = TelefonicaPaletteColor.telefonica_color_telefonicaBlue,
-      controlInverse = TelefonicaPaletteColor.telefonica_color_white,
+      controlInverse = TelefonicaPaletteColor.telefonica_color_grey1,
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/TelefonicaBrandColorsAutogenerated.kt:74]: unrelated token update

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/O2BrandColorsAutogenerated.kt [hunk 14]
```diff
@@ -64,7 +64,7 @@ public object O2BrandColors {
       textLinkSnackbar = O2PaletteColor.o2_color_o2BluePrimary30,
       control = O2PaletteColor.o2_color_grey3,
       controlActivated = O2PaletteColor.o2_color_o2BluePrimary,
-      controlInverse = O2PaletteColor.o2_color_white,
+      controlInverse = O2PaletteColor.o2_color_o2BluePrimary30,
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/O2BrandColorsAutogenerated.kt:67]: unrelated token update

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/BlauBrandColorsAutogenerated.kt [hunk 15]
```diff
@@ -65,7 +65,7 @@ public object BlauBrandColors {
       textLinkSnackbar = BlauPaletteColor.blau_color_blauPurple30,
       control = BlauPaletteColor.blau_color_grey2,
       controlActivated = BlauPaletteColor.blau_color_blauBlueSecondary,
-      controlInverse = BlauPaletteColor.blau_color_white,
+      controlInverse = BlauPaletteColor.blau_color_blauBluePrimary30,
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/BlauBrandColorsAutogenerated.kt:68]: unrelated token update

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/ComponentCatalogActivity.kt [hunk 16]
```diff
@@ -250,6 +252,7 @@ class ComponentCatalogActivity : FragmentActivity() {
             VIVO_NEW -> VivoNewBrand
             TELEFONICA -> TelefonicaBrand
             BLAU -> BlauBrand
+            TU -> TuBrand
```
Comment (yamal-alm) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/ComponentCatalogActivity.kt:255]: adding Tu brand to Catalog

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt [hunk 17]
```diff
@@ -50,6 +51,7 @@ class CatalogMainActivity : AppCompatActivity() {
             ComponentStyle("Vivo New", R.style.MisticaTheme_VivoNew, VIVO_NEW),
             ComponentStyle("Telefonica", R.style.MisticaTheme_Telefonica, TELEFONICA),
             ComponentStyle("Blau", R.style.MisticaTheme_Blau, BLAU),
+            ComponentStyle("Tu", R.style.MisticaTheme_Tu, TU),
```
Comment (yamal-alm) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt:54]: adding Tu brand to Catalog

File: README.md [hunk 18]
```diff
@@ -163,25 +163,7 @@ We have several tokens that are defined by the UX team (colors, radius and text
...
-- As we support XML and Compose, we have to give support to this new token in both systems. For that, we have two main classes [`GenerateXMLFiles`](https://github.com/Telefonica/mistica-android/blob/main/tokens/src/main/java/com/telefonica/mistica/tokens/xml/GenerateXMLFiles.kt) and [`GenerateComposeFiles`](https://github.com/Telefonica/mistica-android/blob/main/tokens/src/main/java/com/telefonica/mistica/tokens/compose/GenerateComposeFiles.kt).  You will have to modify them to add support for the new type of token.
-- To test your development easily, you can download [the tokens files](https://github.com/Telefonica/mistica-design/tree/pre-production/tokens) into `/temp/mistica-design/tokens` directory and run `./gradlew generateMisticaTokens`.
+[There is a more detailed documentation](tokens/README.md) about how to work with this design tokens generation.
```
Comment (yamal-alm) [inline README.md:166]: documentation moved to specific `README.md` file inside tokens module

File: tokens/README.md [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+- Before starting to write code to generate the code and XMLs automatically, you will have to think about how to give support to this new token in Mistica, for example, which attributes you will have to create for example. Once you have this clear, you will be able to start generating that code that could be written manually if this generator didn't exist.
+- As we support XML and Compose, we have to give support to this new token in both systems. For that, we have two main classes [`GenerateXMLFiles`](https://github.com/Telefonica/mistica-android/blob/main/tokens/src/main/java/com/telefonica/mistica/tokens/xml/GenerateXMLFiles.kt) and [`GenerateComposeFiles`](https://github.com/Telefonica/mistica-android/blob/main/tokens/src/main/java/com/telefonica/mistica/tokens/compose/GenerateComposeFiles.kt).  You will have to modify them to add support for the new type of token.
+- To test your development easily, you can download [the tokens files](https://github.com/Telefonica/mistica-design/tree/pre-production/tokens) into `/temp/mistica-design/tokens` directory and run `./gradlew generateMisticaTokens`.
+
+### How to add a new brand
```
Comment (yamal-alm) [inline tokens/README.md:23]: this is the new content, previous sections where just moved from root `README.md` file to this new one

**Review by haynlo — APPROVED** [review]
LGTM, good job!!
The new `How to add a new brand` part is perfectly explained, thanks 🙌🏼

**Review by yceballost — APPROVED** [review]
😍

**Review by juangardi21 — APPROVED** [review]
Nice job 🚀

**Pull request merged**